lib/fog/aws/kinesis.rb

```diff
@@ -0,0 +1,157 @@
+module Fog
+  module AWS
+    class Kinesis < Fog::Service
+      extend Fog::AWS::CredentialFetcher::ServiceMethods
+
+      class LimitExceeded < Fog::Errors::Error; end
+      class ResourceNotFound < Fog::Errors::Error; end
+      class ExpiredIterator < Fog::Errors::Error; end
+      class InvalidArgument < Fog::Errors::Error; end
+      class ProvisionedThroughputExceeded < Fog::Errors::Error; end
+
+      requires :aws_access_key_id, :aws_secret_access_key
+      recognizes :region, :host, :path, :port, :scheme, :persistent, :use_iam_profile, :aws_session_token, :aws_credentials_expire_at, :instrumentor, :instrumentor_name
+
+      request_path 'fog/aws/requests/kinesis'
+
+      request :list_streams
+      request :describe_stream
+      request :create_stream
+      request :delete_stream
+      request :get_shard_iterator
+      request :put_records
+      request :get_records
+
+      class Real
+        include Fog::AWS::CredentialFetcher::ConnectionMethods
+
+        def initialize(options={})
+          @use_iam_profile = options[:use_iam_profile]
+
+          @connection_options = options[:connection_options] || {}
+
+          @instrumentor           = options[:instrumentor]
+          @instrumentor_name      = options[:instrumentor_name] || 'fog.aws.kinesis'
+
+          options[:region] ||= 'us-east-1'
+          @region     = options[:region]
+          @host       = options[:host] || "kinesis.#{options[:region]}.amazonaws.com"
+          @path       = options[:path]        || '/'
+          @persistent = options[:persistent]  || false
```

Comment:
Good to know. It's almost always bit me when I set it to true. It just depends on the usage pattern though. I think, with S3 for instance, persistent works great as long as you don't let it timeout (ie be active at least once per 30 seconds). But with a bit of quiet it dies/explodes. Anyway, something to consider, because the dns/connect/ssl stuff adds up pretty quickly.